include/seqan3/alignment/pairwise/edit_distance_unbanded.hpp

```diff
@@ -430,20 +430,30 @@ struct pairwise_alignment_edit_distance_unbanded
         {
             get<align_result_key::score>(res) = score();
         }
-        if constexpr (std::tuple_size_v<result_type> >= 3)
+
+        if constexpr (std::tuple_size_v<result_type> == 3)
+        {
+            alignment_coordinate end = end_coordinate();
+            get<align_result_key::end>(res) = end;
+        }
+        else if constexpr (std::tuple_size_v<result_type> == 4)
         {
-            get<align_result_key::end>(res) = end_coordinate();
+            alignment_coordinate end = end_coordinate();
+            trace_matrix_type matrix = trace_matrix();
+            std::tuple<size_t, size_t, size_t> stats = alignment_trace_statistics(matrix, end);
+
+            get<align_result_key::end>(res) = end;
+            get<align_result_key::begin>(res) = alignment_begin_coordinate(end, stats);
         }
-        // if constexpr (std::tuple_size_v<result_type> >= 4)
-        // { TODO
-        //     throw std::invalid_argument{"The current output setting is not supported!"};
-        // }
-        if constexpr (std::tuple_size_v<result_type> == 5)
+        else if constexpr (std::tuple_size_v<result_type> == 5)
         {
-            auto && [seq1_trace, seq2_trace] = alignment_trace(database, query, trace_matrix(), end_coordinate());
-            auto & [in1, in2] = get<align_result_key::trace>(res);
-            ranges::copy(seq1_trace, ranges::back_inserter(in1));
-            ranges::copy(seq2_trace, ranges::back_inserter(in2));
+            alignment_coordinate end = end_coordinate();
+            trace_matrix_type matrix = trace_matrix();
+            std::tuple<size_t, size_t, size_t> stats = alignment_trace_statistics(matrix, end);
+
+            get<align_result_key::end>(res) = end;
+            get<align_result_key::begin>(res) = alignment_begin_coordinate(end, stats);
```

Comment:
could be stored already in the contexts before

Reply:
true